.github/workflows/test-and-publish.yml

```diff
@@ -0,0 +1,47 @@
+# Github actions that run tests on each commit, and publish to PyPi on tagged releases
+# build using https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/
+
+name: Test and Release to PyPi
+
+on:
+  push
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+    - name: Install and build
+      run: >-
+        python -m
+        pip install ".[dev]"
+        build
+        --user
+    - name: build binary and tarball
+      run: >-
+        python -m
+        build
+        --sdist
+        --wheel
+        --outdir dist/
+        .
+    - name: test
+      run: >-
+        python -m
+        pip install --upgrade pytest
```

Comment:
Won't we get pytest from the dev dependencies?

Reply:
We would get pytest from dev dependencies if we run setup before getting here. Currently, we don't run setup.

Reply:
I'm not sure I follow this. We are doing `pip install ."[dev]"` in the step before this one and I've checked that this gives me pytest.

```
$pip install ."[dev]"
(...)
$which pytest
/home/jaipreet/miniconda3/envs/burner-12/bin/pytest
```

If we re-install pytest here then we have to keep the version in the GitHub Action and the one in `setup.py` in sync, which is what I want to avoid.

Reply:
Yeah you're totally right, it does run without these lines